core/node/consensus/src/en.rs

```diff
@@ -239,3 +259,31 @@ impl EN {
         Ok(())
     }
 }
+
+/// Wrapper to call [MainNodeClient::fetch_attestation_status] and adapt the return value to [AttestationStatusClient].
+struct MainNodeAttestationStatus(Box<DynClient<L2>>);
+
+#[async_trait]
+impl AttestationStatusClient for MainNodeAttestationStatus {
+    async fn attestation_status(
+        &self,
+        ctx: &ctx::Ctx,
+    ) -> ctx::Result<Option<(attester::GenesisHash, attester::BatchNumber)>> {
+        match ctx.wait(self.0.fetch_attestation_status()).await? {
+            Ok(Some(status)) => {
+                // If this fails the AttestationStatusRunner will log it an retry it later,
+                // but it won't stop the whole node.
+                let status: consensus_dal::AttestationStatus =
+                    zksync_protobuf::serde::deserialize(&status.0)
+                        .context("deserialize(AttestationStatus")?;
+
+                Ok(Some((status.genesis, status.next_batch_to_attest)))
+            }
+            Ok(None) => Ok(None),
+            Err(err) => {
+                tracing::warn!("AttestationStatus call to main node HTTP RPC failed: {err}");
+                Ok(None)
```

Comment:
you can just return the error and it will be logged outside, right?

Reply:
Yes, it would do the same thing. This just followed the error log pattern in another similar API client wrapper in one of the files, can’t remember which

Reply:
I suppose it would be clearer if the Client API didn't return Result<Option<Status>> but rather Result<Status>. Anyway, it doesn't matter much.